crypto/include/internal/dso_conf.h.in

```diff
@@ -10,6 +10,7 @@
 
 #ifndef HEADER_DSO_CONF_H
 # define HEADER_DSO_CONF_H
+# ifndef DSO_NONE
```

Comment:
catch 22: `DSO_NONE` is defined by the perl code below, if `$target{dso_scheme}` is undefined.

This isn't the right way to do this.  If you truly want this, you must make `dso` disablable again, in `Configure`.  You do so by adding it to `@disablables` and removing it from `%deprecated_disablables`, and then you need to change this script as follows:

``` diff
-    my $scheme = uc $target{dso_scheme};
+    my $scheme = $disabled{dso} ? undef : uc $target{dso_scheme};
```

Reply:
That would be compelling, but when I do that the while crypto/dso directory
is no longer built, but I need the stubs in dso_lib.c, which just return not-implemented.

Reply:
This fixes that:

```
diff --git a/Configure b/Configure
index 5a699836f3..cf0ee02e85 100755
--- a/Configure
+++ b/Configure
@@ -1181,7 +1181,7 @@ foreach my $what (sort keys %disabled) {
         my $macro = $disabled_info{$what}->{macro} = "OPENSSL_NO_$WHAT";
 
         if ((grep { $what eq $_ } @{$config{sdirs}})
-                && $what ne 'async' && $what ne 'err') {
+                && $what ne 'async' && $what ne 'err' && $what ne 'dso') {
             @{$config{sdirs}} = grep { $what ne $_} @{$config{sdirs}};
             $disabled_info{$what}->{skipped} = [ catdir('crypto', $what) ];
 
```